include/btu/tex/formats.hpp

```diff
@@ -67,6 +75,7 @@ auto guess_texture_type(std::u8string_view path) noexcept -> std::optional<Textu
 
 auto guess_best_format(DXGI_FORMAT current_format,

```

Comment:
This function is starting to get a lot of parameters. Maybe moving to a struct `GuessBestFormatArgs` would be better, since it would allow us to use designated initializers

Reply:
Refactored, does it look better?